src/fvar.cc

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2018 The OTS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fvar.h"
+
+namespace ots {
+
+// -----------------------------------------------------------------------------
+// OpenTypeFVAR
+// -----------------------------------------------------------------------------
+
+bool OpenTypeFVAR::Parse(const uint8_t* data, size_t length) {
+  Buffer table(data, length);
+  if (!table.ReadU16(&this->majorVersion) ||
+      !table.ReadU16(&this->minorVersion) ||
+      !table.ReadU16(&this->axesArrayOffset) ||
+      !table.ReadU16(&this->reserved) ||
+      !table.ReadU16(&this->axisCount) ||
+      !table.ReadU16(&this->axisSize) ||
+      !table.ReadU16(&this->instanceCount) ||
+      !table.ReadU16(&this->instanceSize)) {
+    return DropVariations("Failed to read table header");
+  }
+  if (this->majorVersion != 1 /*|| this->minorVersion != 0*/) {
+    return DropVariations("Unknown table version");
+  }
+  if (this->axesArrayOffset > length || this->axesArrayOffset < table.offset()) {
```

Comment:
What's wrong with a `this->axesArrayOffset < table.offset()`?

This is the kind of thing I really dislike about OTS.  Making assumptions.

Reply:
Yes, but what's wrong with that? An optimizing compiler can choose to do just that...  I understand it's an stretch and no one does it. But one reason I don't do those kinds of things is because of OTS.

Reply:
Hmm... yes, in principle it could be valid. Not sure how I feel about allowing such tricksy stuff, though -- maybe it seems more likely to indicate a broken tool or corrupted font? @khaledhosny WDYT?

Reply:
Well, I think the idea is to protect faulty implementations from shooting themselves in the foot. I’m not sure if this particular check helps that much, but at least we should make sure all offsets + length are with in the bounds of the table they apply too.